bigquery_etl/query_scheduling/task.py

```diff
@@ -45,6 +45,20 @@ class UnscheduledTask(Exception):
     pass
 
 
+@attr.s(auto_attribs=True)
+class TaskRef:
+    """
+    Representation of a reference to another task.
+
+    The task can be defined in bigquery-etl or in telemetry-airflow.
+    Uses attrs to simplify the class definition and provide validation.
+    Docs: https://www.attrs.org
+    """
+
+    dag_name: str = attr.ib()
+    task_id: str = attr.ib()
+
```

Comment:
We likely have to add an `execution_delta` option here too.

Reply:
Do we also need to perform validation to make sure we don't have duplicate task dependencies listed between the `depends_on` list and automatically inferred dependencies?